File: Programming Basics with JS/05. While Loop - Exercise/oldBooks.js
function solve(input){
    let i = 0;
    let wantedBook = input[i];
    i++;
    let checkedBooks = 0;
    let currentBook = input[i];

    while(currentBook!=="No More Books"){
        if(currentBook === wantedBook){
            console.log(`You checked ${checkedBooks} books and found it.`);
            break;
        } else{
            checkedBooks++;
            i++;
            currentBook=input[i];
        }
    }

    if(currentBook!==wantedBook){
        console.log(`The book you search is not here!`);
        console.log(`You checked ${checkedBooks} books.`);
    }
}

solve(["Bourne",
"True Story",
"Forever",
"More Space",
"The Girl",
"Spaceship",
"Strongest",
"Profit",
"Tripple",
"Stella",
"The Matrix",
"Bourne"])

File: Programming Basics with Java/05 While Loop - Exercise/src/OldBooks.java
import java.util.Scanner;

public class OldBooks {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String book = scan.nextLine();
        int count = 0;
        boolean isFound = false;

        String input = scan.nextLine();

        while(input.equals("No More Books") == false){
            if(book.equals(input)){
                isFound = true;
                break;
            }

            count ++;
            input = scan.nextLine();
        }

        if(isFound){
            System.out.printf("You checked %d books and found it.", count);
        } else {
            System.out.println("The book you search is not here!");
            System.out.printf("You checked %d books.", count);
        }
    }
}